app/src/sharedTest/java/org/oppia/app/parser/HtmlParserTest.kt

```diff
@@ -162,8 +150,61 @@ class HtmlParserTest {
     assertThat(bulletSpan1).isNotNull()
   }
 
-  private fun getResources(): Resources {
-    return ApplicationProvider.getApplicationContext<Context>().resources
+  @Test
+  fun testHtmlContent_textLeadingMarginSpan_isAdded() {
+    val textView =
+      activityTestRule.activity.findViewById(R.id.test_html_content_text_view) as TextView
+    val htmlParser = htmlParserFactory.create(/* entityType= */ "", /* entityId= */
+      "", /* imageCenterAlign= */
+      true
+    )
+    val htmlResult: Spannable = htmlParser.parseOppiaHtml(
+      """
+            <ul>
```

Comment:
I think this list could be simplified and still test the same patterns. Consider:

```xml
<ul>
  <li>Item 1</li>
  <li>Item 2</li>
  <li>Numbered list:
    <ol>
      <li>Nested item in numbered list 1</li>
      <li>Nested item in numbered list 2</li>
    </ol>
  </li>
  <li>Nested list:
    <ul>
      <li>Double nested list:
        <ul>
          <li>
            Double nested item
          </li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
```

That being said, could we break this into smaller pieces and have targeted tests for each piece? E.g. a test for an unordered list just needs a single unordered list, and ditto for an ordered list. Then you could have a single test for one level of nesting, another for multiple levels of nested, and others for various error cases.

Reply:
I have created `TagHandlerTest` in `Utility` module please check if that is correct.